src/domain/executor/profile_generator.py

```diff
@@ -0,0 +1,152 @@
+import functools
```

Comment:
Just as a call out, my PR has inflight work on the profile generator that needs review. I also am considering the notion of creating another Profile Generator class and separating out the synchronous case from constant and poisson, since they are pretty exclusive. It makes a bit more sense in the context of the multiple rates that can be passed in since we iterate through the profiles based off of how many rates are passed in which doesn't really align with the synchronous use case. I don't understand if there are any other considerations that would impact that decision so it is sort of an open question on my PR

Reply:
Could you please send a link to that PR with the code? Can't find it

Reply:
Sure!
https://github.com/neuralmagic/guidellm/pull/2/files